### ISSUE TITLE
[MuonGun] NaturalRateInjector test failure (Trac #2184)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2176]
39/461 Test  #39: MuonGun::test_NaturalRateInjector.py ...........................***Failed    0.81 sec
INFO (MuonGun): The table path GaisserH4a_atmod12_SIBYLL does not exist! Assuming you meant /build/buildslave/stheno/Ubuntu_18_04/build/MuonGun/resources/tables/GaisserH4a_atmod12_SIBYLL... (__init__.py:53 in load_model)
FATAL (MuonGun): The provided energy distribution is not normalized (integrates to 2.40583e-12). This could be because the minimum and maximum energies were changed from their defaults. (NaturalRateInjector.cxx:116 in void I3MuonGun::NaturalRateInjector::SetEnergyDistribution(I3MuonGun::EnergyDistributionPtr))
Traceback (most recent call last):
  File "/build/buildslave/stheno/Ubuntu_18_04/source/MuonGun/resources/test/test_NaturalRateInjector.py", line 19, in <module>
    generator = nevents*MuonGun.NaturalRateInjector(surface, flux, energies)
RuntimeError: The provided energy distribution is not normalized (integrates to 2.40583e-12). This could be because the minimum and maximum energies were changed from their defaults. (in void I3MuonGun::NaturalRateInjector::SetEnergyDistribution(I3MuonGun::EnergyDistributionPtr))

Comment by jvs on slack:  "the problem is that the energy bounds are now set correctly from the bounds of support of the energy spline, and at the upper end of the support, the energy pdf is quite tiny. as a result, the numerical integration in linear space fails hard. "

Migrated from https://code.icecube.wisc.edu/ticket/2184
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:13", 
    "description": "\n 39/461 Test  #39: MuonGun::test_NaturalRateInjector.py ...........................***Failed    0.81 sec\nINFO (MuonGun): The table path GaisserH4a_atmod12_SIBYLL does not exist! Assuming you meant /build/buildslave/stheno/Ubuntu_18_04/build/MuonGun/resources/tables/GaisserH4a_atmod12_SIBYLL... (__init__.py:53 in load_model)\nFATAL (MuonGun): The provided energy distribution is not normalized (integrates to 2.40583e-12). This could be because the minimum and maximum energies were changed from their defaults. (NaturalRateInjector.cxx:116 in void I3MuonGun::NaturalRateInjector::SetEnergyDistribution(I3MuonGun::EnergyDistributionPtr))\nTraceback (most recent call last):\n  File \"/build/buildslave/stheno/Ubuntu_18_04/source/MuonGun/resources/test/test_NaturalRateInjector.py\", line 19, in <module>\n    generator = nevents*MuonGun.NaturalRateInjector(surface, flux, energies)\nRuntimeError: The provided energy distribution is not normalized (integrates to 2.40583e-12). This could be because the minimum and maximum energies were changed from their defaults. (in void I3MuonGun::NaturalRateInjector::SetEnergyDistribution(I3MuonGun::EnergyDistributionPtr))\n\n\nComment by jvs on slack:  \"the problem is that the energy bounds are now set correctly from the bounds of support of the energy spline, and at the upper end of the support, the energy pdf is quite tiny. as a result, the numerical integration in linear space fails hard. \"\n\n", 
    "reporter": "olivas", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067313248429", 
    "component": "combo core", 
    "summary": "[MuonGun] NaturalRateInjector test failure", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-08-24T18:45:26", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "task"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180827T09:22:32: 
* jvansanten changed status from "new" to "closed"
* jvansanten changed resolution from "" to "fixed"
* jvansanten commented:

> In changeset 165001:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="165001"
> Integrate bundle fluxes over log(energy) for better numerical stability. Fixes #2176
> 
> Take log(energy) instead of energy in EnergyDistribution::GetLog(), perform integration in this space directly. Along the way, replace bound member functions with lambdas for readability.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:13: 
* olivas changed milestone from "Summer Solstice 2018" to ""
* olivas commented:

> Milestone Summer Solstice 2018 deleted
